src/models.rs

```diff
@@ -313,6 +350,22 @@ pub struct ClassifiedGeneratedTextResult {
     pub input_tokens: Option<Vec<GeneratedToken>>,
 }
 
+/// The response format of the /api/v1/text/task/generation-detection endpoint
+#[derive(Default, Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
+pub struct GenerationWithDetectionResult {
```

Comment:
Agreed!
I had it like that because I was comparing structs with their already existing counterparts for the `/classification-with-text-generation` endpoint. With the initial implementation done, it does make sense to move related classes together. Done.